### PR TITLE
Provide Route.toFunction as alternative to Route.asyncHandler

### PR DESCRIPTION
**Pull request by raboof**
On top of #3114

### CODE REVIEW

File: akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala [hunk 1]
```diff
@@ -73,8 +73,10 @@ object Route {
                                 exceptionHandler: ExceptionHandler         = null): Flow[HttpRequest, HttpResponse, NotUsed] =
     Flow[HttpRequest].mapAsync(1)(asyncHandler(route))
 
-  def asyncHandler(route: Route, routingSettings: RoutingSettings, parserSettings: ParserSettings)(implicit system: ClassicActorSystemProvider): HttpRequest => Future[HttpResponse] = {
+  def toFunction(route: Route)(implicit system: ClassicActorSystemProvider): HttpRequest => Future[HttpResponse] = {
```
Comment (jrudolph) [inline akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala:76]: Is this just about naming?

Comment (raboof) [reply akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala:76]: No: akka-grpc was using the `asyncHandler` that takes a `Route` and a bunch of implicits, and had to replace it with this method that required explicit RoutingSettings and ParserSettings. Removing those, however, would clash with the existing asyncHandler.

Renaming seemed elegant (with `toFunction` being similar to `toFlow`).

Alternatively we could change the deprecated `asyncHandler` and turn some implicit parameters into explicit ones, but that does look a bit strange. I have no strong preference.

Comment (jrudolph) [reply akka-http/src/main/scala/akka/http/scaladsl/server/Route.scala:76]: Ah, I see, thanks for the explanation.

I find `toFunction` a bit unexpected as `Route` already is an alias for a function. But maybe people don't even care about that (and most people wouldn't use it anyway). So, fine for me!

File: docs/src/test/scala/docs/http/scaladsl/server/directives/JsonStreamingFullExamples.scala [hunk 2]
```diff
@@ -7,6 +7,7 @@ package docs.http.scaladsl.server.directives
 import com.github.ghik.silencer.silent
 import org.scalatest.wordspec.AnyWordSpec
 
+@silent("The outer reference in this type test cannot be checked at run time")
```
Comment (jrudolph) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/JsonStreamingFullExamples.scala:10]: Still needed?

**Pull request merged**